bravado/client.py

```diff
@@ -102,11 +102,23 @@ def from_url(cls, spec_url, http_client=None, request_headers=None,
         :param config: Config dict for bravado and bravado_core.
             See CONFIG_DEFAULTS in :module:`bravado_core.spec`.
             See CONFIG_DEFAULTS in :module:`bravado.client`.
+
+        :rtype: :class:`bravado_core.spec.Spec`
         """
         log.debug(u"Loading from %s" % spec_url)
         http_client = http_client or RequestsClient()
         loader = Loader(http_client, request_headers=request_headers)
         spec_dict = loader.load_spec(spec_url)
+
+        # RefResolver may have to download additional json files (remote refs)
+        # via http. Wrap http_client's request() so that request headers are
+        # passed along with the request transparently. Yeah, this is not ideal,
+        # but since RefResolver has new found responsibilities, it is
+        # functional.
```

Comment:
1) yes
2) bravado-core if spec validation is disabled; ssv if spec validation is enabled. Downloads only occur once because the resolver is shared between ssv and bravado-core.
3) What would a "clone of this http_client" look like in code? Why?